docs/user/reference/action.md

```diff
@@ -3,19 +3,49 @@
 
 > See also: {ref}`manage-actions`
 
-In Juju, an **action** is a script that is triggered via the {ref}`juju-cli` and applied to a {ref}`unit <unit>`. It contains a list of commands defined by a  {ref}`charm <charm>` to allow a {ref}`user <user>` with the right {ref}`access level <user-access-levels>` to interact with an {ref}`application <application>` in ways specific to the application. This may include anything from creating a snapshot of a database, adding a user to a system, dumping debug information, etc.
+An action is used to perform a named, parameterised operation and report back the results of said operation.
+It is triggered via the {ref}`juju-cli` and applied to one or more {ref}`units <unit>`.
+
+Actions are defined by a  {ref}`charm <charm>` to allow a {ref}`user <user>` with the right {ref}`access level <user-access-levels>` to interact with an {ref}`application <application>` in ways specific to the application.
+This may include anything from creating a snapshot of a database, adding a user to a system, dumping debug information, etc.
 
 > See examples: [Charmhub | `kafka` > Actions](https://charmhub.io/kafka/actions), [Charmhub | `prometheus-k8s` > Actions](https://charmhub.io/prometheus-k8s/actions), etc.
 
-<!--UPDATES IN JUJU V.3.0:
-https://discourse.charmhub.io/t/new-feature-in-juju-2-8-improved-actions-experience/3182
+The default behaviour is that the {ref}`juju-cli` blocks and waits for the action to complete. This synchronous behaviour allows actions to be easily included in command-line pipelines.
+As an action is executing, any progress messages as reported by the action are logged to the terminal. When the action completes, the result is printed.
+An action result is a map of key values, containing data set by the action as it runs, plus the overall exit code
+of the action process itself, as well as the content of stdout and stderr.
+
+The execution of an action is organised into {ref}`tasks <task>` and {ref}`operations <operation>`.
+(If an action defines a named unit of work -- e.g., back up the database -- that can be executed on selected units, a task is the execution of the action on each target unit, and an operation is the group of tasks queued by running an action across one or more units.)
+
+The code used to implement an action can call the following hook commands in addition to {ref}`the others <list-of-hook-commands>`:
+* `action-log`: to report a progress message
+* `action-get`: to get the value of a named action parameter as supplied by the user
+* `action-set`: to set a value in the action results map
+* `action-fail`: to mark the action as failed along with a failure message
+
```

Comment:
We also talked about moving the contents of the action command pages from here https://canonical-juju.readthedocs-hosted.com/en/3.6/user/reference/hook-commands/list-of-hook-commands/ to this action page. If you want, I can take care of that myself.

Reply:
Since that involves changes to `conf.py` and `hook-commands/main.go` (I assume), I was thinking we'd do that as a separate PR and take care of any other required changes in that area at the same time.